Feature/upgrade to java 17

ivaylomitrev opened the pull request:
The PR bumps the Noark extraction validator to Java 17. This generally means:
* Java 8 --> Java 17 upgrade
* Javax --> Jakarta migration
* dependency and plugin version upgrades - Apache commons, Apache POI, others

I have also moved away from PDFBox for PDF/A validation and have substituted that with VeraPDF as PDFBox proved difficult to migrate to Java 17 especially when we want to use the project in library mode internally.

dkumanov commented:
The readme documentation still mentions JDK 8 as required.

dkumanov commented on noark-extraction-validator/pom.xml:
```diff
@@ -28,22 +28,21 @@
-			<version>1.2</version>
-		</dependency>
-
 		<!-- XJC (XSD scema converter) -->
```
```suggestion
		<!-- XJC (XSD schema converter) -->
```

dkumanov commented on noark-extraction-validator/src/main/java/com/documaster/validator/config/properties/InternalProperties.java:
```diff
@@ -41,8 +44,13 @@ public InternalProperties(List<String> propertyFiles) throws IOException, Config
 
 			try (InputStream is = getClass().getClassLoader().getResourceAsStream(file)) {
```
We no longer need the stream it looks like. We can get rid of the `throws IOException` declaration for the constructors as well.

ivaylomitrev replied:
Changed w/ https://github.com/documaster/noark-extraction-validator/pull/66/commits/b58f7551f05889b80d1e9f6a529710525dfa8533

dkumanov replied:
We still need to remove the throws IOException declaration in the `Noark5Properties` class constructor

ivaylomitrev replied:
Removed w/ https://github.com/documaster/noark-extraction-validator/pull/66/commits/5817337587bc9bf4934ee24bdfaedff585796dad

ivaylomitrev commented:
> The readme documentation still mentions JDK 8 as required.

Updated with https://github.com/documaster/noark-extraction-validator/pull/66/commits/963d6ea65b6f835580ca1edc5a866b542d02a67e

I also updated the debian configuration with https://github.com/documaster/noark-extraction-validator/pull/66/commits/10694360a567cafcc17209d3b9778be31d0039ba

ivaylomitrev commented:
Comments this far have been addressed.

petterreinholdtsen commented:
When do you plan to include this in the develop or master branches?

dkumanov commented on noark-extraction-validator/pom.xml:
```diff
@@ -77,35 +76,23 @@
-			<artifactId>pdfbox</artifactId>
-		</dependency>
-
         <!--- Apache Preflight -->
```
```suggestion
        <!--- Apache VeraPDF -->
```

dkumanov commented on noark-extraction-validator/src/main/java/com/documaster/validator/validation/utils/PDFAValidator.java:
```diff
@@ -45,37 +54,16 @@ public static String getFileType(File file) throws IOException {
 
 	public static boolean isValidPdfaFile(File file) throws IOException {
```
Again no longer needed.
```suggestion
	public static boolean isValidPdfaFile(File file) {
```

ivaylomitrev replied:
Applied w/ https://github.com/documaster/noark-extraction-validator/pull/66/commits/bad998fa5c1d4df46e42c6aa1d9f47d6d84ae284.

As part of the same commit I also extracted the InputStream to its own statement in the try-with-resources so that we free the resources properly

dkumanov commented on noark-extraction-validator/pom.xml:
```diff
@@ -28,22 +28,21 @@
+		<!-- XJC (XSD schema converter) -->
+
 		<dependency>
-			<groupId>commons-logging</groupId>
```
This package is needed for running the project as a standalone jar tool.

ivaylomitrev replied:
Addressed w/ https://github.com/documaster/noark-extraction-validator/pull/66/commits/b94c3fa94913345f901788b827972196eca60618

dkumanov commented on noark-extraction-validator/pom.xml:
```diff
@@ -54,13 +53,13 @@
-			<groupId>commons-configuration</groupId>
-			<artifactId>commons-configuration</artifactId>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-configuration2</artifactId>
```
This new version of commons-configuration seam to require this package as well:
```
		<dependency>
			<groupId>commons-beanutils</groupId>
			<artifactId>commons-beanutils</artifactId>
			<version>1.9.4</version>
		</dependency>
```
otherwise we will hit the following issue when running the standalone jar:
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/beanutils/BeanIntrospector
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:375)
        at jdk.proxy2/jdk.proxy2.$Proxy9.<clinit>(Unknown Source)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1053)
        at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1039)
        at org.apache.commons.configuration2.builder.fluent.Parameters.createParametersProxy(Parameters.java:243)
        at org.apache.commons.configuration2.builder.fluent.Parameters.properties(Parameters.java:198)
        at com.documaster.validator.config.properties.InternalProperties.<init>(InternalProperties.java:46)
        at com.documaster.validator.config.properties.Noark5Properties.<init>(Noark5Properties.java:41)
        at com.documaster.validator.config.commands.Noark5Command.getProperties(Noark5Command.java:164)
        at com.documaster.validator.validation.Noark5Validator.validateStructure(Noark5Validator.java:211)
        at com.documaster.validator.validation.Noark5Validator.run(Noark5Validator.java:90)
        at com.documaster.validator.main.Run.main(Run.java:51)
Caused by: java.lang.ClassNotFoundException: org.apache.commons.beanutils.BeanIntrospector
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        ... 18 more
```

ivaylomitrev replied:
Addressed w/ https://github.com/documaster/noark-extraction-validator/pull/66/commits/b94c3fa94913345f901788b827972196eca60618

dkumanov reviewed (approved):
Everything looks okay and as far as I tested works without issues.

ivaylomitrev commented:
> When do you plan to include this in the develop or master branches?

@petterreinholdtsen , I will merge it this week in develop now that it has been approved, but I cannot provide a timeline for merging it in master yet.

ivaylomitrev commented:
Squashed in preparation for merging. No diff after squashing
